sites/frontend/lib/parse-capi/index.js

```diff
@@ -19,17 +19,42 @@ type props = {
 
 const defaultArgs = { config: {}, contentFields: {} };
 
-export default ({ contentFields, config }: props = defaultArgs) => ({
-    // here we create our own object of CAPI content on the 'CAPI' key
-    CAPI: {
-        headline: headline(get(config, 'page.headline', '')),
-        standfirst: standfirst(get(contentFields, 'fields.standfirst', '')),
-        main: main(get(contentFields, 'fields.main', '')),
-        body: body(
-            get(contentFields, 'fields.blocks.body', [])
-                .map(block => block.bodyHtml)
-                .filter(Boolean)
-                .join(''),
-        ),
-    },
-});
+// TODO: we shouldn't do any cleaning or data transformation here, but
+// instead pass through the required data exactly. Dotcom Rendering
+// should be just that - a rendering layer only.
+//
+// We *should* however validate the data we receive at this point.
+export default ({ contentFields, config }: props = defaultArgs) => {
+    const nav = get(config, 'nav', []);
+    const pillars = nav.pillars.map(link => {
```

Comment:
It looks like the default/fallback value of `nav` is an empty array `[]`. If `nav` ever gets assigned this default value then `nav.pillars.map` will throw an error something along the lines of "can't call map of undefined". 

We could either set the default value to `{pillars: []}` Or we could just use a ternary for this assignment where we check for the existence of `nav.pillars` and if `undefined` assign `[]` to `pillars`.